src/org.xtuml.bp.core/src/org/xtuml/bp/core/common/PersistableModelComponent.java

```diff
@@ -629,7 +629,7 @@ public void persistSelfAndChildren() throws CoreException {
  * @throws CoreException
  */
     public void load(IProgressMonitor monitor) throws CoreException {
-        if (status == STATUS_LOADING) {
+        if (status == STATUS_LOADING && status == PersistableModelComponent.STATUS_LOADING) {
```

Comment:
In PMC.load(...) why did you have to add the && status == PersistableModelComponent.STATUS_LAODING? It looks redundant to me.